Kudu.Core/Jobs/ContinuousJobLogger.cs

```diff
@@ -26,21 +26,29 @@ public ContinuousJobLogger(string jobName, IEnvironment environment, ITraceFacto
 
         public void ResetLockedStatusFile()
         {
+            FileStream resetLockedStatusFile;
             try
             {
-                if (_lockedStatusFile != null)
-                {
-                    _lockedStatusFile.Dispose();
-                }
+                resetLockedStatusFile = File.Open(GetStatusFilePath(), FileMode.OpenOrCreate, FileAccess.Read, FileShare.ReadWrite);
+            }
+            catch (UnauthorizedAccessException)
+            {
+                // This most likely means we still have the lock
+                return;
```

Comment:
throw if _lockedStatusFile is null.

Reply:
Yeah, sorry I missed that. Updating right now.

Reply:
I don't really know what the context of Suwat comment above is, but I don't see a throw if `_lockedStatusFile` is null

Reply:
Right, so @suwatch initially pointed out that if we get an `UnauthorizedAccessException` and the `_lockedStatusFile` is null, that would be a problem.
Then, I discussed with Suwat offline that the lock we are acquiring is not exclusive (it allows sharing for read and write) and we came to the conclusion that we should not hit the case of `UnauthorizedAccessException` and should just make a simple change to acquire a new lock and replace the old one.